query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/MigrationTest.java

```diff
@@ -233,36 +269,166 @@ public void testQueryWithDistinctCountAggregation() {
   }
 
   @Test
-  public void testQueryWithContainsKeyValueOperator() {
+  public void testQueryWithEQFilterForMapAttribute() {
     Builder builder = QueryRequest.newBuilder();
-    Expression spanTag = createSimpleAttributeExpression("Span.tags").build();
+    Expression spanTag = createComplexAttributeExpression("Span.tags", "FLAGS").build();
```

Comment:
does our code lowercase this? I would imagine we should respect the key's casing, unless we're lowercasing the data as we persist it.

Reply:
This is interesting, need to check. But, it seems that as prior existing tests (for CONTAINS_KEYVALUE) were also changed to lower case.
- https://github.com/hypertrace/query-service/blob/main/query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverterTest.java#L574

Reply:
Yeah, we should follow up on this. May be working correctly, but it caught my eye. If we're pushing everything to lowercase on ingestion side, we have to do this; if we're not, I'd consider it a bug.

Reply:
No, we are not doing anything specific for lowercase. It is the assertion utility in the test were written as lowercase for comparing two string for expected and actual string.

- https://github.com/hypertrace/query-service/blob/main/query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/MigrationTest.java#L347

Reply:
Ugh, we should remove that. Case matters in (certain positions of) a query, and this is testing the query itself.